infer/environment.cc

```diff
@@ -1090,7 +1090,7 @@ core::TypePtr Environment::processBinding(core::Context ctx, const cfg::CFG &inW
 
                 const core::TypeAndOrigins &ty = getAndFillTypeAndOrigin(ctx, c->value);
                 ENFORCE(c->cast != core::Names::uncheckedLet());
-                if (c->cast != core::Names::cast()) {
+                if (c->cast == core::Names::let() || c->cast == core::Names::assertType()) {
```

Comment:
There's a finite set of valid names here.

I find this logic easier to read when they are explicitly listed.